redbot/core/core_commands.py

```diff
@@ -3613,10 +3620,15 @@ async def _set_serverprefix(self, ctx: commands.Context, *prefixes: str):
             - `[p]set serverprefix "! "` - Quotes are needed to use spaces in prefixes.
             - `[p]set serverprefix "@[botname] "` - This uses a mention as the prefix.
             - `[p]set serverprefix ! ? .` - Sets multiple prefixes.
+            - `[p]set serverprefix guild_name ? - Sets prefix for specified guild.
 
         **Arguments:**
             - `[prefixes...]` - The prefixes the bot will respond to on this server. Leave blank to clear server prefixes.
         """
+        if specified_guild is not None:
+            # prefer specified guild over ctx.guild when provided as argument
+            ctx.guild = specified_guild
```

Comment:
Instead of overriding the `ctx` attribute (which affects *anything* using that context object), do something more like
```py
if server is None:
    server = ctx.guild
```
then replace anywhere the command currently uses `ctx.guild` with `server`.